src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -463,6 +471,106 @@ export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
     }))
   }
 
+  const prepareVoteAuxiliaryData = (
+    hwStakeSigningFile: HwSigningData,
+    votingPublicKeyHex: string,
+    addressParameters: _AddressParameters,
+    nonce: BigInt,
+  ): LedgerTypes.TxAuxiliaryData => ({
+    type: LedgerTypes.TxAuxiliaryDataType.TUPLE,
+    params: {
+      metadata: {
+        type: LedgerTypes.TxMetadataType.CATALYST_REGISTRATION,
+        params: {
+          votingPublicKeyHex,
+          stakingPath: hwStakeSigningFile.path,
+          rewardsDestination: {
+            type: addressParameters.addressType,
+            params: {
+              spendingPath: addressParameters.paymentPath,
+              stakingPath: addressParameters.stakePath,
+            },
+          },
+          nonce: `${nonce}`,
+        },
+      },
+    },
+  })
+
+  const prepareDummyInput = (): LedgerTypes.TxInput => ({
+    txHashHex: '0'.repeat(64),
+    outputIndex: 0,
+    path: parseBIP32Path('1852H/1815H/0H/0/0'),
+  })
+
+  const prepareDummyOutput = (): LedgerTypes.TxOutput => ({
+    destination: {
+      type: LedgerTypes.TxOutputDestinationType.DEVICE_OWNED,
+      params: {
+        type: LedgerTypes.AddressType.BASE,
+        params: {
+          spendingPath: parseBIP32Path('1852H/1815H/0H/0/0'),
+          stakingPath: parseBIP32Path('1852H/1815H/0H/2/0'),
+        },
+      },
+    },
+    amount: '1',
+  })
+
+  const prepareDummyTx = (
+    network: LedgerTypes.Network, auxiliaryData: LedgerTypes.TxAuxiliaryData,
+  ): LedgerTypes.SignTransactionRequest => ({
+    signingMode: LedgerTypes.TransactionSigningMode.ORDINARY_TRANSACTION,
+    tx: {
+      network,
+      inputs: [prepareDummyInput()],
+      outputs: [prepareDummyOutput()],
+      fee: 1,
+      ttl: 1,
+      certificates: null,
+      withdrawals: null,
+      auxiliaryData,
+      validityIntervalStart: null,
+    },
+  })
+
+  const signVotingRegistrationMetaData = async (
+    auxiliarySigningFiles: HwSigningData[],
+    hwStakeSigningFile: HwSigningData,
+    paymentAddressBech32: string,
+    votePublicKeyHex: VotePublicKeyHex,
+    network: Network,
+    nonce: BigInt,
+  ): Promise<CborHex> => {
+    const { data: address } : { data: Buffer } = bech32.decode(paymentAddressBech32)
+    const addressParams = getAddressParameters(auxiliarySigningFiles, address, network)
+    if (!addressParams || !addressParams.stakePath || addressParams.address.compare(address)) {
```

Comment:
we don't really need to force the stakePath to be supplied in addressParams - Ledger "cares" only about the address being given as parameters and that the type must not be Byron.

Reply:
I'm not sure if I understand this correctly. I'm using the `addressParams.stakePath` to find **hwsfile** with same **path** and then use it's `cborXPubKeyHex` to break it down into **stake pub hex** which is part of metadata fed to ledger.

So if I don't check it here, then I'm not able to do this mapping.

Reply:
Fixed